webapp/packages/core-sdk/src/Resource/ResourceAlias.ts

```diff
@@ -15,25 +15,72 @@ export type ResourceAliasOptions = Readonly<Record<ResourceAliasOptionsKey, Reso
 export abstract class ResourceAlias<TKey, TOptions extends ResourceAliasOptions> {
   readonly id: string;
   readonly options: TOptions;
+  target: any;
+  parent?: ResourceAlias<TKey, any>;
   private readonly typescriptHack: TKey;
   abstract readonly name: string;
 
-  constructor(id: string, options: TOptions) {
+  constructor(id: string, options: TOptions, parent?: ResourceAlias<TKey, any>) {
     this.typescriptHack = null as any;
     this.id = id;
     this.options = options;
+    this.parent = parent;
+  }
+
+  find<TOptions extends ResourceAliasOptions = any>(
+    key: ResourceAlias<TKey, TOptions> | ResourceAliasFactory<TKey, TOptions>,
+  ): ResourceAlias<TKey, TOptions> | undefined {
+    if (this.id === key.id) {
+      return this as any;
+    }
+
+    if (this.parent) {
+      return this.parent.find(key);
+    }
+
+    return undefined;
+  }
+
+  setTarget(target: any): this {
+    this.target = target;
+    return this;
+  }
+
+  setParent(parent: ResourceAlias<TKey, any>): this {
+    const copy = new (this.constructor as any)(this.id, this.options, parent) as this;
+    return copy.setTarget(this.target);
   }
 
   isEqual(key: ResourceAlias<TKey, any>): boolean {
-    if (isResourceAlias(key)) {
-      return key.id === this.id && isObjectsEqual(this.options, key.options);
+    if (!!this.parent !== !!key.parent) {
```

Comment:
Looks strange, do we need double negation here?

Reply:
you are right, fixed